aten/src/ATen/test/basic.cpp

```diff
@@ -173,32 +173,32 @@ std::cout << (a == 10.) << " -- should be 1" << std::endl;
 #endif
 */
 
-void TestAddingAValueWithScalar(Type& type) {
-  Tensor a = rand({4, 3}, type);
-  ASSERT_TRUE((ones({4, 3}, type) + a).equal(add(a, 1)));
+void TestAddingAValueWithScalar(TensorOptions &options) {
+  Tensor a = rand({4, 3}, options);
+  ASSERT_TRUE((ones({4, 3}, options) + a).equal(add(a, 1)));
 }
 
-void TestSelect(Type& type) {
-  Tensor a = rand({3, 7}, type);
+void TestSelect(TensorOptions &options) {
+  Tensor a = rand({3, 7}, options);
   auto a_13 = select(a, 1, 3);
   auto a_13_02 = select(select(a, 1, 3), 0, 2);
   ASSERT_TRUE(a[0][3].equal(a_13[0]));
   ASSERT_TRUE(a[2][3].equal(a_13_02));
 }
 
-void TestZeroDim(Type& type) {
-  Tensor a = at::scalar_tensor(4, type.options()); // rand(type, {1});
+void TestZeroDim(TensorOptions &options) {
```

Comment:
I don't think you need references here all over this file -- the issue was you can't have a value-type Type.